utils/saveDB.js

```diff
@@ -107,27 +114,65 @@ async function findTimeDifferenceToPrompt(client, channelId, referenceMessage) {
 
 async function saveDB(client, channelId) {
     try {
-        const messagesList = await fetchMessagesUntilPrompt(client, channelId);
+        const messagesList = await fetchImageMessagesUntilPrompt(client, channelId);
         const messagesData = [];
 
         for (const message of messagesList) {
             const numOfReactions = countReactions(message);
             const imageLink = getImageLinkFromMessage(message);
             const timeToPost = await findTimeDifferenceToPrompt(client, channelId, message);
-
+            const message_id = message.id;
             const messageData = {
-                numOfReactions: numOfReactions,
+                message_id: message_id,
+                num_reactions: numOfReactions,
                 response_image: imageLink,
-                timeToPost: timeToPost
+                time_to_respond: timeToPost
             };
             messagesData.push(messageData);
         }
-        // TODO: Save messagesData to database here
+        // Insert each messageData into the database
+        for (const messageData of messagesData) {
+            await insertResponseData(messageData);
+        }
+
+        console.log('All data has been saved to the database');
         return messagesData;
     } catch (error) {
         console.error('Error in saveDB:', error);
         throw error;
     }
 }
 
+async function insertResponseData(messageData) {
+    const connection = await pool.getConnection();
+    try {
+        // First, check if a record with the same message_id already exists
+        const checkQuery = 'SELECT * FROM responses WHERE message_id = ?';
+        const [existingRecords] = await connection.query(checkQuery, [messageData.message_id]);
+
+        // If no existing record is found, proceed to insert the new record
+        if (existingRecords.length === 0) {
+            const insertQuery = `
+                INSERT INTO responses (response_image, num_reactions, time_to_respond, message_id) 
+                VALUES (?, ?, ?, ?)
+            `;
+            const values = [
+                messageData.response_image,
+                messageData.num_reactions,
+                messageData.time_to_respond,
+                messageData.message_id
+            ];
+            await connection.query(insertQuery, values);
```

Comment:
you can actively use pool.query in place of this and connection release

Reply:
do you want me to make line 165 this? `await pool.query(insertQuery, values)`

Reply:
yeah, and ditch connection.release